SourceX/DiabloUI/dialogs.cpp

```diff
@@ -215,12 +216,16 @@ void Deinit()
 		UnloadTtfFont();
 	if (textInputWasActive)
 		SDL_StartTextInput();
-	for (std::size_t i = 0; i < dialogItemsSize; ++i) {
-		dialogItems[i].FreeCache();
+
+	for (std::size_t i = 0; i < vecOkDialog.size(); i++) {
+		UiItemBase *pUIItem = vecOkDialog[i];
+		if (pUIItem)

```

Comment:
`delete NULL;` is safe

Generally, when checking for nullptr it's better to be explicit (`if (pUIItem != NULL)`) because implicit boolean conversion operator can be overriden.